File: campus-trade-server/src/main/java/com/campustrade/entity/Banner.java
package com.campustrade.entity;

import com.campustrade.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class Banner extends BaseEntity {

    private String title;
    private String subtitle;
    private String imageUrl;
    private String linkUrl;
    private String bgColor;
    private String buttonText;
    private Integer sortOrder;
    private Integer status;
}

File: campus-trade-user/src/api/banner.ts
import request from '@/utils/request'

export interface BannerVO {
  id: number
  title: string
  subtitle: string
  imageUrl: string
  linkUrl: string
  bgColor: string
  sortOrder: number
  status: number
  createTime: string
}

export const getActiveBanners = () => request.get<never, BannerVO[]>('/banner/active')